Downsample/Downsample.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+Import a ZDF point cloud and downsample it.
+*/
+
+#include <Zivid/CloudVisualizer.h>
+#include <Zivid/Zivid.h>
+#include <Eigen/Dense>
+#include <Eigen/Core>
+
+#include <iostream>
+Eigen::MatrixXf sumline(Eigen::MatrixXf&, int, int);
+Eigen::MatrixXf gridsum(Eigen::MatrixXf&, int);
+float myround(float); 
+float isnanmask(float);
+float nantozero(float);
+Zivid::PointCloud downsample(Zivid::PointCloud&, int);
+
+int main()
+{
+	try
+	{
+		Zivid::Application zivid;
+
+		std::string Filename = "Zivid3D.zdf";
+		std::cout << "Reading " << Filename << " point cloud" << std::endl;
+		Zivid::Frame frame = Zivid::Frame(Filename);
+
+		auto pointCloud = frame.getPointCloud();
+		auto downSamplingFactor = 4;
+
+		Zivid::PointCloud pointCloudDownsampled(pointCloud.height()/downSamplingFactor, pointCloud.width()/downSamplingFactor);
+
+		pointCloudDownsampled = downsample(pointCloud, downSamplingFactor);
+
+		std::cout << "Setting up visualization" << std::endl;
+		{
```

Comment:
Then I need to use Zivid::Application zivid; and Zivid::CloudVisualizer vis; in the function?

Reply:
I think you need to create the `CloudVisualizer` inside the function, just as you create it inside the scope now. At least if I'm correct that you're recreating it since you can't reuse it?

And then you would need to take `zivid` by reference as a parameter to the function.

Reply:
done